synapse/tests/test_lib_aha.py

```diff
@@ -1310,3 +1371,201 @@ async def test_aha_provision_listen_dns_name(self):
+            purl00 = await aha.addAhaSvcProv('0.cell')
+            purl01 = await aha.addAhaSvcProv('1.cell', provinfo={'mirror': '0.cell'})
+
+            cell00 = await aha.enter_context(self.getTestCell(conf={'aha:provision': purl00}))
+            cell01 = await aha.enter_context(self.getTestCell(conf={'aha:provision': purl01}))
+
+            await cell01.sync()
+
+            async def sleep99(cell):
+                await cell.boss.promote('sleep99', cell.auth.rootuser)
+                await cell00.fire('sleep99')
+                await asyncio.sleep(99)
+
+            async with cell00.waiter(2, 'sleep99', timeout=2):
+                task00 = cell00.schedCoro(sleep99(cell00))
+                task01 = cell01.schedCoro(sleep99(cell01))
+
+            proxy = await aha.enter_context(aha.getLocalProxy())
+            tasks = [task async for task in cell00.getTasks(timeout=3)]
+            self.len(2, tasks)
+            self.eq(tasks[0]['service'], '0.cell.synapse')
+            self.eq(tasks[1]['service'], '1.cell.synapse')
+
+            self.eq(tasks[0], await cell00.getTask(tasks[0].get('iden')))
+            self.eq(tasks[1], await cell00.getTask(tasks[1].get('iden')))
+            self.none(await cell00.getTask(tasks[1].get('iden'), peers=False))
+
+            self.true(await cell00.killTask(tasks[0].get('iden')))
+
+            task01 = tasks[1].get('iden')
+            self.false(await cell00.killTask(task01, peers=False))
+
+            self.true(await cell00.killTask(task01))
+
+            self.none(await cell00.getTask(task01))
+            self.false(await cell00.killTask(task01))
+
+            self.none(await cell00.getAhaProxy(feats=(('newp', 9),)))
+
+    async def test_aha_req_proxy(self):
+
+        async with self.getTestAha() as aha:
+
+            info = {
+                'urlinfo': {
+                    'host': 'localhost',
+                    'port': 12345,
+                    'scheme': 'ssl',
+                    'user': 'root',
+                },
+                'online': 'fakesession',
+                'ready': True,
+            }
+
+            await aha.addAhaSvc('testsvc.aha', info)
+            svcdef = await aha.getAhaSvc('testsvc.aha')
+            with self.raises(s_exc.NotReady):
+                await aha.reqAhaSvcProxy(svcdef)
+
+    async def test_lib_aha_peer_api(self):
+
+        async with self.getTestAha() as aha:
+
+            purl00 = await aha.addAhaSvcProv('0.cell')
+            purl01 = await aha.addAhaSvcProv('1.cell', provinfo={'mirror': '0.cell'})
+
+            cell00 = await aha.enter_context(self.getTestCell(conf={'aha:provision': purl00}))
+            cell01 = await aha.enter_context(self.getTestCell(conf={'aha:provision': purl01}))
+
+            await cell01.sync()
+
+            todo = s_common.todo('getCellInfo')
+            items = [item async for item in cell00.callPeerApi(todo)]
+            self.len(1, items)
```

Comment:
Shouldn't this be 2 items? One for each cell?

Reply:
No because I'm calling the todo on a single cell. Had it been on the aha then yes 2!

Reply:
The skiprun=self.runid used by `callPeerApi()` causes AHA to skip our self. Had there been 3, we should have expected 2 :)

Reply:
Can we add a third cell to this test to validate that then?

Reply:
will be in the next commit